addons/pos_loyalty/models/pos_order.py

```diff
@@ -52,6 +52,27 @@ def validate_coupon_programs(self, point_changes, new_codes):
             'payload': {},
         }
 
+    def add_loyalty_history_lines(self, coupon_data, coupon_updates):
+        id_mapping = {item['old_id']: int(item['id']) for item in coupon_updates}
+        history_lines_to_create = []
+        for coupon in coupon_data:
+            if int(coupon['card_id']) not in id_mapping:
+                continue
+            card_id = id_mapping[int(coupon['card_id'])]
+            issued = coupon['won']
+            cost = coupon['spent']
+            if (issued or cost) and card_id > 0:
```

Comment:
In pos_loyalty for this case:
there is no existing loyalty card for a customer
there is a sale quotation for this customer
go to a pos session and settle the order
this will create a card from both sides sale_loyalty and pos_loyalty
so for this case i believe that they delete the first generated card and for this card the id_mapping will have -ve card_id